### PR TITLE
Feature/party invite popup

### PR DESCRIPTION
**Pull request by samuelWilliams99**
image

Also adds F3 to toggle mouse input and fixes a bug with players joining after parties created

### CODE REVIEW

File: gamemode/modules/party/sv_party.lua [hunk 1]
```diff
@@ -367,10 +369,16 @@ end )
...
-    
+
     local spawnPos = raft:GetSpawnPosition( ply )
 
     ply:SetPos( spawnPos )
 end )
+
+net.Receive( "RVR_Party_requestFullUpdate", function( len, ply )
+    net.Start( "RVR_Party_requestFullUpdate" )
+    net.WriteTable( party.parties )
```
Comment (plally) [inline gamemode/modules/party/sv_party.lua]: werent we indenting net messages?

**Review by plally — CHANGES_REQUESTED** [review]

File: gamemode/modules/party/cl_party.lua [hunk 2]
```diff
@@ -2,6 +2,20 @@ RVR.Party = RVR.Party or {}
...
+hook.Add( "PlayerButtonDown", "RVR_Party_ToggleMouse", function( ply, key )
+    -- This hook gets called like a million times
+    if CurTime() - lastClick < 0.05 then return end
+    lastClick = CurTime()
+
+    if key ~= KEY_F3 then return end
+
+    mouseEnabled = not mouseEnabled
+    gui.EnableScreenClicker( mouseEnabled )
+end )
```
Comment (plally) [inline gamemode/modules/party/cl_party.lua]:
```lua
hook.Add( "PlayerBindPress", "", function( ply, bind, pressed )
    if not ( pressed and bind == "gm_showspare1" ) then return end
    -- stuff
end )
```

**Review by plally — APPROVED** [review]

File: gamemode/modules/party/cl_party_invite_popup.lua [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    label:SetTextColor( color )
+    label:SizeToContentsX()
+    label:SetTall( ScrH() * 0.02 )
+    label:SetPos( 0, y - ScrH() * 0.01 )
+    label:CenterHorizontal()
+
+    return label
+end
+
+local function makeButton( text, x, y, h, frame, f )
```
Comment (overlordbleck) [inline gamemode/modules/party/cl_party_invite_popup.lua]: f > func

File: gamemode/modules/party/cl_party_invite_popup.lua [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        RunConsoleCommand( "rvr", "party_join", partyData.id )
+    end )
+    makeButton( "Ignore", w * 0.85, buttonY, buttonSectionH * 0.6, frame, function()
+        blacklist[partyData.id] = true
+    end )
+
+    timer.Simple( GAMEMODE.Config.Party.INVITE_COOLDOWN, function()
+        frame:Remove()
+    end )
+end
```
Comment (overlordbleck) [inline gamemode/modules/party/cl_party_invite_popup.lua:135]: balloon pop

File: gamemode/modules/party/sv_party.lua [hunk 5]
```diff
@@ -367,10 +369,16 @@ end )
 hook.Add( "RVR_SuccessfulPlayerSpawn", "RVR_Party_Raft_Spawn", function( ply )
     local partyData = ply:GetParty()
     if not partyData then return end -- This should never happen, but just to be sure.
-    
+
     local raft = RVR.getRaft( partyData.raftID )
-    
+
```
Comment (overlordbleck) [inline gamemode/modules/party/sv_party.lua]:
```suggestion
```

personal preference, I like to keep my variables in one spot

**Review by overlordbleck — APPROVED** [review]
🎈

File: gamemode/modules/party/cl_party_invite_popup.lua (hunk 3 above)
Comment (samuelWilliams99) [reply gamemode/modules/party/cl_party_invite_popup.lua]: I agree, f is better than func

**Pull request merged**